egui/src/widgets/plot/mod.rs

```diff
@@ -966,43 +1093,34 @@ impl PreparedPlot {
 
         let font_id = TextStyle::Body.resolve(ui.style());
 
-        let base: i64 = 10;
-        let basef = base as f64;
-
-        let min_line_spacing_in_points = 6.0; // TODO: large enough for a wide label
-        let step_size = transform.dvalue_dpos()[axis] * min_line_spacing_in_points;
-        let step_size = basef.powi(step_size.abs().log(basef).ceil() as i32);
-
-        let step_size_in_points = (transform.dpos_dvalue()[axis] * step_size).abs() as f32;
-
         // Where on the cross-dimension to show the label values
+        let bounds = transform.bounds();
         let value_cross = 0.0_f64.clamp(bounds.min[1 - axis], bounds.max[1 - axis]);
 
-        for i in 0.. {
-            let value_main = step_size * (bounds.min[axis] / step_size + i as f64).floor();
-            if value_main > bounds.max[axis] {
-                break;
-            }
+        let bounds = (bounds.min[axis], bounds.max[axis]);
+        let bounds_frame_ratio = transform.dvalue_dpos()[axis] * MIN_LINE_SPACING_IN_POINTS;
+
+        let input = GridInput {
+            bounds,
+            bounds_frame_ratio,
+        };
```

Comment:
minimize the amount of bindings:

```suggestion
        let input = GridInput {
            bounds: (bounds.min[axis], bounds.max[axis]),
            bounds_frame_ratio: transform.dvalue_dpos()[axis] * MIN_LINE_SPACING_IN_POINTS,
        };
```

Reply:
Done.